pkg/apis/openebs.io/upgrade/v1alpha1/upgraderesult.go

```diff
@@ -20,76 +20,123 @@ import (
-	FailedTaskCount int `json:"failedTaskCount"`
-	// Tasks represents the list of tasks
+// UpgradeResource represents a resource that needs to
+// be upgraded to a desired version
+type UpgradeResource struct {
+	// Name of the resource to be upgraded
+	Name string `json:"name"`
+	// Kind is the type of resource i.e.
+	// PVC, SPC, ..
+	Kind string `json:"kind"`
+	// APIVersion of the resource
+	APIVersion string `json:"apiVersion"`
+	// Namespace of the resource
+	Namespace string `json:"namespace"`
+	// BeforeUpgrade represents the state of the
+	// related resources before upgrade i.e. for a
+	// PV, related resources could be cvr, target
+	// deployment, target svc, etc.
+	BeforeUpgrade []ResourceState `json:"beforeUpgrade"`
+	// AfterUpgrade represents the state of the
+	// related resources after upgrade
+	AfterUpgrade []ResourceState `json:"afterUpgrade"`
+	// Tasks are the runtasks that needs to be
+	// executed to perform this upgrade
 	Tasks []Task `json:"tasks"`
 }
 
-// Task represents an upgrade task and its subtasks details
+// Task represents details of a task(runtask) required
+// to be executed for upgrading a particular resource
 type Task struct {
 	// Name of the task
 	Name string `json:"name"`
-	// TotalSubTaskCount is the no of subtasks
-	// i.e. runtasks that needs to be executed
-	// to complete a task
-	TotalSubTaskCount int `json:"totalSubTaskCount"`
-	// CompletedSubTaskCount is the no of subtasks
-	// that has successfully executed
-	CompletedSubTaskCount int `json:"completedSubTaskCount"`
-	// FailedSubTaskCount is the no of subtasks that has
-	// failed
-	FailedSubTaskCount int `json:"failedSubTaskCount"`
-	// LastCompletedSubTaskID is the id of the last successfully
-	// completed runtask of a particular task
-	LastCompletedSubTaskID string `json:"lastCompletedSubTaskID"`
+	// Status is the status of the task which
+	// could be successful or failed
+	Status string `json:"status"`
+	// LastTransitionTime is the last time the status
+	// transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+	// Message is a human readable message
+	// indicating details about the task
+	Message string `json:"message"`
+	// LastError is the last error occurred
+	// while executing this task
+	LastError string `json:"lastError"`
+	// StartTime of the task
+	StartTime *metav1.Time `json:"startTime"`
+	// EndTime of the task
+	EndTime *metav1.Time `json:"endTime"`
+	// Retries is the no of times this task
+	// has tried to execute
+	Retries int `json:"retries"`
+}
+
+// ResourceState represents the state of a resource
+type ResourceState struct {
+	// Name of the resource
+	Name string `json:"name"`
+	// Kind is the type of resource i.e.
+	// cvr, deployment, ..
+	Kind string `json:"kind"`
+	// APIVersion of the resource
+	APIVersion string `json:"apiVersion"`
+	// Namespace of the resource
+	Namespace string `json:"namespace"`
+	// Status of the resource
+	Status string `json:"status"`
```

Comment:
Does this status also needs to be a type struct with more details ?

Reply:
not needed as of now